src/libxtp/gwbse/sigma_exact.cc

```diff
@@ -0,0 +1,117 @@
+/*
+ *            Copyright 2009-2019 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "votca/xtp/vc2index.h"
+#include <votca/xtp/rpa.h>
+#include <votca/xtp/sigma_exact.h>
+#include <votca/xtp/threecenter.h>
+
+namespace votca {
+namespace xtp {
+
+void Sigma_Exact::PrepareScreening() {
+  _rpa_solution = _rpa.Diagonalize_H2p();
+  _residues = CalcResidues();
```

Comment:
Can we reconsider calculating the residues on the fly as they again increase the memory footprint? Where they bigger or smaller than the eigenvectors?